addons/account/models/account_move.py

```diff
@@ -2147,13 +2147,19 @@ def post(self):
                 move.company_id.account_bank_reconciliation_start = move.date
 
         for move in self:
-            if not move.partner_id: continue
-            if move.type.startswith('out_'):
+            if move.is_sale_document() and move.journal_id.sale_activity_type_id:
+                move.activity_schedule(
+                    date_deadline=min(move.line_ids.mapped('date_maturity') or [move.date]),
```

Comment:
you were right. It doesn't work because `move.line_ids.mapped('date_maturity')` could return a list containing `False` values...

Reply:
`[date for date in move.line_ids.mapped('date_maturity') if date]` then :disappointed: